pdr_backend/lake/alt_bar.py

```diff
@@ -0,0 +1,158 @@
+import logging
+from typing import List, Union
+import numpy as np
+
+from enforce_typing import enforce_types
+import pandas as pd
+
+# add config pair: volume_threshold in ppss
+
+
+@enforce_types
+def _extract_bars(data: pd.DataFrame, metric: str, threshold: float = 50000):
+    """
+    For loop which compiles the various bars: dollar, volume, or tick.
+
+    We did investigate the use of trying to solve this in a vectorised manner but found that a For loop worked well.
+
+    :param data: raw ohlcv data from kaiko Contains 6 columns - timestamp, open, high, low, close and volume.
+    :param metric: cum_ticks, cum_dollar_value, cum_volume
+    :param threshold: A cumulative value above this threshold triggers a sample to be taken.
+    :return: The financial data structure with the updated start timestamp since last bar.
+    """
+
+    list_bars = []
+    cache: list = []
```

Comment:
please rename this variable something like "bars_cache" or another more relevant word. `cache` is often used when dealing with external cache in Python projects, so it is confusing to the reader that this is in fact a custom cache, a list for the purpose of local processing